lib/kubeclient/oidc_auth_provider.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Kubeclient
+  # Uses OIDC id-tokens and refreshes them if they are stale.
+  class OIDCAuthProvider
+    class OpenIDConnectDependencyError < LoadError # rubocop:disable Lint/InheritException
+    end
+
+    class << self
+      def token(auth_provider)
```

Comment:
[optional nit, your call] The name `auth_provider` sounds like this would contain the ['auth-provider'] part of the config, but it contains the ['auth-provider']['config'] sub-part.  Maybe `config`? `auth_provider_config` (ouch)?  `provider_config`?

Reply:
Good point, I've gone with `provider_config`.